gap/matrix/classical.gi

```diff
@@ -759,26 +759,26 @@ RECOG.TestRandomElement := function (recognise, grp)
             Info(InfoClassical,2,o1, " ", o2);
             ### ACN March 2007 needed Projective order
             if  q mod 2 = 0 and (o1 <= 2 or o2 <= 2) then
-                return fail;
+                return TemporaryFailure;
             elif q mod 2 <> 0 and (o1 <= 4 or o2 <= 4) then
-                return fail;
+                return TemporaryFailure;
             fi;
             if ( (q+1) mod o1 = 0 and (q+1) mod o2 = 0) then
                 if not IsPowerOfPrime(q+1,2) and q <> 8 then
                     if  not ForAny( PrimeDivisors(o1), r ->
                       IsPrimitivePrimeDivisor(p,2*recognise.a,r)) then
-                        return false;
+                        return TemporaryFailure;
                     fi;
                     if not ForAny( PrimeDivisors(o2), r ->
                       IsPrimitivePrimeDivisor(p,2*recognise.a,r)) then
-                        return false;
+                        return TemporaryFailure;
                     fi;
                 elif q = 8 then
                     if not o1 mod 9 = 0 or not o2 mod 9 = 0 then
-                        return false;
+                        return TemporaryFailure;
```

Comment:
@aniemeyer so here for example: we are looking for two elements of order 9; if only find one, and the other is of order 3, then in the previous version of the code we ended here with `false = NeverApplicable` and so recognition overall failed. That seems wrong; with the changes in this PR, more naming tests pass. The only drawback I can think of is "we might try too hard in some cases", but that's it. Can you confirm?